### PR TITLE
Add new Dialog component

### PR DESCRIPTION
**Pull request by cristian-seoane**
Added a new Dialog component based on  MaterialUI 's Modal component

### CODE REVIEW

File: src/components/Dialog/dialogStories.styles.tsx [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { css } from '@emotion/react';
```
Comment (cristian-seoane) [inline src/components/Dialog/dialogStories.styles.tsx:1]: Brandon: I'm not sure about this file, please tell me if I should move it to another directory or put these styles into .stories file.

File: src/components/Dialog/dialogHeader.types.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+export default interface DialogHeaderProps {
+    title?: string;
```
Comment (BrandonDenham) [inline src/components/Dialog/dialogHeader.types.ts:2]: dont think this is optional

File: src/components/Dialog/dialogStories.styles.tsx (hunk 1 above)
Comment (BrandonDenham) [reply src/components/Dialog/dialogStories.styles.tsx:1]: should at least create a subdirectory

**Review by BrandonDenham — APPROVED** [review]

File: src/components/Dialog/dialogHeader.types.ts (hunk 2 above)
Comment (cristian-seoane) [reply src/components/Dialog/dialogHeader.types.ts:2]: ok so, it should be:    title: required, subtitle:optional  right?  or both required?

File: src/components/Dialog/dialogStories.styles.tsx (hunk 1 above)
Comment (cristian-seoane) [reply src/components/Dialog/dialogStories.styles.tsx:1]: yeap, so  dialog.stories and dialogStories.styles  files should be in:  dialog/stories/    right?

File: src/components/Dialog/dialogHeader.types.ts (hunk 2 above)
Comment (BrandonDenham) [reply src/components/Dialog/dialogHeader.types.ts:2]: title required, subtitle optional

File: src/components/Dialog/dialogStories.styles.tsx (hunk 1 above)
Comment (BrandonDenham) [reply src/components/Dialog/dialogStories.styles.tsx:1]: ya if possible

**Review by BrandonDenham — APPROVED** [review]

File: src/components/Dialog/dialogStories.styles.tsx (hunk 1 above)
Comment (cristian-seoane) [reply src/components/Dialog/dialogStories.styles.tsx:1]: done!

File: src/components/Dialog/dialogHeader.types.ts (hunk 2 above)
Comment (cristian-seoane) [reply src/components/Dialog/dialogHeader.types.ts:2]: done!

**Pull request merged**